lib/eight_ball.rb

```diff
@@ -85,9 +103,28 @@ def self.with(name, parameters = {})
     yield if enabled? name, parameters
   end
 
+  # Marshalls the {EightBall::Feature Features}. This can be useful for
+  # converting the data to, e.g., a JSON file.
+  #
+  # If a {EightBall::Marshallers Marshaller} is provided, use it.
+  #
+  # If no {EightBall::Marshallers Marshaller} is provided, uses the same
+  # Marshaller that the Provider is configured with.
+  #
+  # If the {EightBall::Providers Provider} does not expose a
+  # {EightBall::Marshallers Marshaller}, this will default to the
+  # {EightBall::Marshallers::Json JSON Marshaller}.
+  def self.marshall(marshaller = nil)
+    marshaller ||=
+      (provider.class.method_defined?(:marshaller) && provider.marshaller) ||
```

Comment:
> (provider.class.method_defined?(:marshaller) && provider.marshaller)

First time seeing. Any advantages over `provider.try(:marshaller)`?

Reply:
(Just to address before the edit, in case: `try` is part of Rails).

I _think_ your way is better for dealing with dynamically created methods, so I'll change it to `respond_to?`. But honestly not sure there is any advantage/disadvantage to either method.